src/mapReducers.js

```diff
@@ -40,7 +41,7 @@ function calculateNewStateForAllKeys(state, action, resolveReducer) {
     return reducer(stateForKey, action);
```

Comment:
Could you just call `calculateNewStateForKey` here for each key?

Reply:
I don't use `eq`, I use `import shallowEqual from 'shallowequal';`

Ok, I will reuse that method. Also, I will add the possibility that key extractor returns an array of keys.